app/views/contact_infos/confirm.html.erb

```diff
@@ -1,6 +1,13 @@
 <%= page_heading "Email Verification" %>
 <% if @handler_result.errors.any? %>
-  <p>Sorry, we couldn't verify an email using the confirmation code you provided.  If you typed it in manually, please check for errors.  If you're still having trouble, you might try to send the confirmation email again.</p>
+  <p>Sorry, we couldn't verify an email using the verification code you provided.  If you typed it in manually, please check for errors.  If you're still having trouble, you might try to send the verification email again.</p>
+
+  <% if signed_in? %>
+    <p><%= button_to 'Resend Verification',
+                     resend_confirmation_contact_info_path(current_user.email_addresses.first),
+                     method: :put,
+                     class: 'standard' %></p>
+  <% end %>
```

Comment:
I know I asked you to do this, but it doesn't actually make sense b/c

1. there might not be a `current_user` (login not required to confirm an email address, which is correct)
2. The email they are trying to confirm might not be their first email.  

Can you instead make the message read:

> Sorry, we couldn't verify an email using the verification code you provided.  If you typed it in manually, please check for errors.  If you're still having trouble, please resend the verification email from your user profile.

... and make "your user profile" be a link to their profile page on the email tab.  If they are logged in, the link will take them right there.  If they aren't logged in, they'll be asked to log in.